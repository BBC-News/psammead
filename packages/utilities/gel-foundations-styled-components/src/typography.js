import { css } from 'styled-components';
import { MEDIA_QUERY_TYPOGRAPHY } from '@bbc/gel-constants/breakpoints';

export const GEL_LONG_PRIMER = css`
  font-size: 0.9375em;
  line-height: 1.125rem;
  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 0.875em;
  }
`;

/* Font styling below is a subset of BBC GEL Typography "Body Copy" */
export const GEL_BODY_COPY = css`
  font-size: 0.9375em;
  line-height: 1.25rem;
  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_AND_LARGER} {
    font-size: 1em;
    line-height: 1.375rem;
  }
`;

export const GEL_BREVIER = css`
  font-size: 0.875em;
  line-height: 1rem;
  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    line-height: 1.125rem;
  }
  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 0.8125em;
  }
`;

/* Font styling below is a subset of BBC GEL Typography "Canon" */
export const GEL_CANON = css`
  font-size: 1.75em;
  line-height: 2rem;
  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 2em;
    line-height: 2.25rem;
  }
  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 2.75em;
    line-height: 3rem;
  }
`;

/* Font styling below is a subset of BBC GEL Typography "Trafalgar" */
export const GEL_TRAFALGAR = css`
  font-size: 1.25em;
  line-height: 1.5rem;
  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 1.5em;
    line-height: 1.75rem;
  }
  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 2em;
    line-height: 2.25rem;
  }
`;

export const GEL_MINION = css`
  font-size: 0.75em;
  line-height: 1em;
`;
