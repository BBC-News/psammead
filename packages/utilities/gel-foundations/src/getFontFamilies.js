const fontFamilies = {
  news: {
    serif: `ReithSerif, Helvetica, Arial, sans-serif`,
    sans: `ReithSans, Helvetica, Arial, sans-serif`,
    sansCond: `ReithSansCondensed, Helvetica, Arial, sans-serif`,
  },
  persian: {
    serif: `NassimPersian, Arial, Verdana, Geneva, Helvetica, sans serif`,
    sans: `NassimPersian, Arial, Verdana, Geneva, Helvetica, sans serif`,
    sansCond: `NassimPersian, Arial, Verdana, Geneva, Helvetica, sans serif`,
  },
  igbo: {
    serif: `Helmet, Freesans, Helvetica, Arial, sans-serif`,
    sans: `Helmet, Freesans, Helvetica, Arial, sans-serif`,
    sansCond: `Helmet, Freesans, Helvetica, Arial, sans-serif`,
  },
  pidgin: {
    serif: `Helmet, Freesans, Helvetica, Arial, sans-serif`,
    sans: `Helmet, Freesans, Helvetica, Arial, sans-serif`,
    sansCond: `Helmet, Freesans, Helvetica, Arial, sans-serif`,
  },
  yoruba: {
    serif: `Helmet, Freesans, Helvetica, Arial, sans-serif`,
    sans: `Helmet, Freesans, Helvetica, Arial, sans-serif`,
    sansCond: `Helmet, Freesans, Helvetica, Arial, sans-serif`,
  },
};

export const getSerif = service =>
  fontFamilies[service] ? fontFamilies[service].serif : null;

export const getSans = service =>
  fontFamilies[service] ? fontFamilies[service].sans : null;

export const getSansCond = service =>
  fontFamilies[service] ? fontFamilies[service].sansCond : null;
