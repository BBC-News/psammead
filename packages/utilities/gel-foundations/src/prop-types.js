import scriptPropType from './propTypes/scripts';

export { scriptPropType };
