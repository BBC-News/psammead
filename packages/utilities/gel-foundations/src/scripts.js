export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
