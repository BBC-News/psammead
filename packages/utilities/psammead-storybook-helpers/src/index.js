import withServicesKnob from './withServicesKnob';
import buildRTLSubstories from './buildRTLSubstories';
import LANGUAGE_VARIANTS from './text-variants';

export { LANGUAGE_VARIANTS, withServicesKnob, buildRTLSubstories };
