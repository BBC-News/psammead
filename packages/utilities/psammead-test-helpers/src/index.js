import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ShallowRenderer from 'react-test-renderer/shallow';

export const shouldMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const shallowRender = component => {
  const shallowRenderer = new ShallowRenderer();
  shallowRenderer.render(component);
  const result = shallowRenderer.getRenderOutput();

  return result;
};

export const shouldShallowMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = shallowRender(component);
    expect(tree).toMatchSnapshot();
  });
};

export const isNull = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toBeNull();
  });
};

const errorIfMissingKey = (keys, object, message) => {
  keys.forEach(key => {
    if (!(key in object)) {
      throw new Error(`Missing value '${key}' in ${message}.`);
    }
  });
};

const checkKeysExistInBothObjects = (object1, object2, message1, message2) => {
  const object1Keys = Object.keys(object1);
  const object2Keys = Object.keys(object2);

  errorIfMissingKey(object1Keys, object2, message2);
  errorIfMissingKey(object2Keys, object1, message1);
};

const checkTypesOfExports = (
  actualExportsByName,
  actualExports,
  expectedExports,
  utilityName,
) => {
  actualExportsByName.forEach(actualExportName => {
    const actualExportValue = actualExports[utilityName][actualExportName];
    const expectedExport = expectedExports[utilityName][actualExportName];
    const typeCheck = typeof actualExportValue === expectedExport; // eslint-disable-line valid-typeof

    // if this fails it is likely that an export is missing from the unit test expectation
    expect(typeCheck).toBe(true);
  });
};

export const testUtilityPackages = (
  actualExports,
  expectedExports,
  packageName,
) => {
  const actualUtilities = Object.keys(actualExports);

  // check if the actual exported file has defined expected values to match and that the expected values are actually exported
  checkKeysExistInBothObjects(
    actualExports,
    expectedExports,
    `the actual utilities for '${packageName}'`,
    `the expected utilities for '${packageName}'`,
  );

  actualUtilities.forEach(utilityName => {
    const actualExportsByName = Object.keys(actualExports[utilityName]);

    // check if each of the actual exported consts have an expected value to match and that the expected values are actually exported
    checkKeysExistInBothObjects(
      actualExports[utilityName],
      expectedExports[utilityName],
      `the actual export for '${packageName}/${utilityName}'`,
      `the expected export for '${packageName}/${utilityName}'`,
    );

    checkTypesOfExports(
      actualExportsByName,
      actualExports,
      expectedExports,
      utilityName,
    );
  });
};
