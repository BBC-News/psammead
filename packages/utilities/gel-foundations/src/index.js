import * as breakpoints from './breakpoints';
import * as propTypes from './prop-types';
import * as scripts from './scripts';
import * as spacings from './spacings';
import * as typography from './typography';

export { breakpoints };
export { propTypes };
export { scripts };
export { spacings };
export { typography };
