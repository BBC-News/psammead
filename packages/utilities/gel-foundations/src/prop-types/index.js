/* eslint-disable import/prefer-default-export */

import scripts from './scripts';

export { scripts };
