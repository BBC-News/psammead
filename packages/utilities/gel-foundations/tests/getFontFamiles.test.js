import { getSerif, getSans, getSansCond } from '../src/getFontFamilies';

describe('getFontFamilies', () => {
  describe('getSerif', () => {
    it('Returns news serif font family', () => {
      expect(getSerif('news')).toMatchSnapshot();
    });
  });
  describe('getSans', () => {
    it('Returns news sans font family', () => {
      expect(getSans('news')).toMatchSnapshot();
    });
  });
  describe('getSansCond', () => {
    it('Returns news sans condensed font family', () => {
      expect(getSansCond('news')).toMatchSnapshot();
    });
  });
});
