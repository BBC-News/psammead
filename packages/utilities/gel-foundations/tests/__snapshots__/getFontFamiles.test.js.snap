// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFontFamilies getSans Returns news sans font family 1`] = `"ReithSans, Helvetica, Arial, sans-serif"`;

exports[`getFontFamilies getSansCond Returns news sans condensed font family 1`] = `"ReithSansCondensed, Helvetica, Arial, sans-serif"`;

exports[`getFontFamilies getSerif Returns news serif font family 1`] = `"ReithSerif, Helvetica, Arial, sans-serif"`;
